src/transformers/models/lightglue/modeling_lightglue.py

```diff
@@ -0,0 +1,1149 @@
+        current_states = encoder_hidden_states if is_cross_attention else hidden_states
+        attention_mask = encoder_attention_mask if is_cross_attention else attention_mask
+        if attention_mask is not None:
+            attention_mask = attention_mask.squeeze()
+            attention_mask = torch.where(attention_mask == -0.0, 1, 0)
+
+        key_layer = self.transpose_for_scores(self.key(current_states))
+        value_layer = self.transpose_for_scores(self.value(current_states))
+        if sinusoidal_pos is not None:
+            if self.rotary_value:
+                query_layer, key_layer, value_layer = self.apply_rotary_position_embeddings(
+                    sinusoidal_pos, query_layer, key_layer, value_layer
+                )
+            else:
+                query_layer, key_layer = self.apply_rotary_position_embeddings(sinusoidal_pos, query_layer, key_layer)
+
+        query_layer = query_layer.reshape(batch_size, q_len, self.num_attention_heads, self.attention_head_size)
+        key_layer = key_layer.reshape(batch_size, q_len, self.num_attention_heads, self.attention_head_size)
+        value_layer = value_layer.reshape(batch_size, q_len, self.num_attention_heads, self.attention_head_size)
+
+        input_dtype = query_layer.dtype
+        if input_dtype == torch.float32:
+            if torch.is_autocast_enabled():
+                target_dtype = torch.get_autocast_gpu_dtype()
+            # Handle the case where the model is quantized
+            elif hasattr(self.config, "_pre_quantization_dtype"):
+                target_dtype = self.config._pre_quantization_dtype
+            else:
+                target_dtype = self.query.weight.dtype
+
+            logger.warning_once(
+                f"The input hidden states seems to be silently casted in float32, this might be related to"
+                f" the fact you have upcasted embedding or layer norm layers in float32. We will cast back the input in"
+                f" {target_dtype}."
+            )
+
+            query_layer = query_layer.to(target_dtype)
+            key_layer = key_layer.to(target_dtype)
+            value_layer = value_layer.to(target_dtype)
+
+        attn_output = _flash_attention_forward(
+            query_layer,
+            key_layer,
+            value_layer,
+            attention_mask,
+            q_len,
+            is_causal=False,
+        )
+
+        attn_output = attn_output.reshape(batch_size, q_len, self.all_head_size).contiguous()
+        outputs = (attn_output,)
+        return outputs
+
+
+class LightGlueMLP(nn.Module):
+    def __init__(self, config: LightGlueConfig):
+        super().__init__()
+        embeddings_dim = config.descriptor_dim
+        self.dense = nn.Linear(2 * embeddings_dim, 2 * embeddings_dim)
+        self.layer_norm = nn.LayerNorm(2 * embeddings_dim, elementwise_affine=True)
+        self.activation = ACT2FN[config.hidden_act]
+        self.output = nn.Linear(2 * embeddings_dim, embeddings_dim)
```

Comment:
this looks classique and we usually define intermediate dim for the dimension of the intermediate MLP.